switcheroo/data_store/data_stores.py

```diff
@@ -78,6 +78,5 @@ def retrieve(self, host: str, user: str) -> str:
         try:
             with open(key_path, mode="rt", encoding="utf-8") as key_file:
                 return key_file.read()
-        except FileNotFoundError:
-            raise KeyNotFoundException()
-
+        except FileNotFoundError as exc:
+            raise KeyNotFoundException() from exc
```

Comment:
you could add some details here indicating which key was not found (think from the perspective of someone getting this error)

Reply:
Updated